src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogQueuePublisher.java

```diff
@@ -70,23 +72,56 @@ protected void doRun() throws Exception {
             Map<String, Set<String>> tags = DatadogUtilities.getTagsFromGlobalTags();
             // Add JenkinsUrl Tag
             tags = TagsUtil.addTagToTags(tags, "jenkins_url", DatadogUtilities.getJenkinsUrl());
-
             long size = 0;
             long buildable = queue.countBuildableItems();
             long pending = queue.getPendingItems().size();
             long stuck = 0;
             long blocked = 0;
+            String hostname = DatadogUtilities.getHostname(null);
             final Queue.Item[] items = queue.getItems();
             for (Queue.Item item : items) {
+                Map<String, Set<String>> job_tags = DatadogUtilities.getTagsFromGlobalTags();
+                job_tags = TagsUtil.addTagToTags(job_tags, "jenkins_url", DatadogUtilities.getJenkinsUrl());
+                
+                String job_name;
+                Task task = item.task;
+                if (task instanceof Run){ 
+                    job_name = ((Run)task).getParent().getFullName();
+                } else {
+                    job_name = task.getFullDisplayName();
```

Comment:
I think we should be more careful here. I'd rather not tag (or actually set the job_name to unknown) rather than tagging by something we have no control on.
`Task` may be extended by any plugin, and the full display name can totally be a unique identifier which will create context explosion.

Reply:
That makes sense, we would not want that. I will look into it more

Reply:
So, I think checking for `Run` might also be a problem if this is the case. What do you think about going back to this suggestion: https://github.com/jenkinsci/datadog-plugin/pull/73#discussion_r438617085

Reply:
I think that seems like the right approach :)

Reply:
Updated!